exec/java-exec/src/main/java/org/apache/drill/exec/store/pcap/decoder/Packet.java

```diff
@@ -0,0 +1,371 @@
+    } else {
+      return 0;
+    }
+  }
+
+  public long getTimestamp() {
+    return timestamp;
+  }
+
+  public int getPacketLength() {
+    return packetLength;
+  }
+
+  public InetAddress getSrc_ip() {
+    return getIPAddress(true);
+  }
+
+  public InetAddress getDst_ip() {
+    return getIPAddress(false);
+  }
+
+  public String getEthernetSource() {
+    return getEthernetAddress(PacketConstants.ETHER_SRC_OFFSET);
+  }
+
+  public String getEthernetDestination() {
+    return getEthernetAddress(PacketConstants.ETHER_DST_OFFSET);
+  }
+
+  public int getSequenceNumber() {
+    if (isTcpPacket()) {
+      int sequenceOffset = PacketConstants.ETHER_HEADER_LENGTH + getIPHeaderLength() + getTCPHeaderLength(raw) + 4;
+      return Math.abs(convertInt(raw, sequenceOffset));
+    } else {
+      return 0;
+    }
+  }
+
+  public int getSrc_port() {
+    if (isPPPoV6Packet()) {
+      return getPort(64);
+    }
+    if (isIpV6Packet()) {
+      if (isRoutingV6) {
+        return getPort(136);
+      }
+      return getPort(40);
+    }
+    return getPort(0);
+  }
+
+  public int getDst_port() {
+    if (isPPPoV6Packet()) {
+      return getPort(66);
+    }
+    if (isIpV6Packet()) {
+      if (isRoutingV6) {
+        return getPort(138);
+      }
+      return getPort(42);
+    }
+    return getPort(2);
+  }
+
+  public byte[] getData() {
+    int payloadDataStart = getIPHeaderLength();
+    if (isTcpPacket()) {
+      payloadDataStart += this.getTCPHeaderLength(raw);
+    } else if (isUdpPacket()) {
+      payloadDataStart += this.getUDPHeaderLength();
+    } else {
+      return null;
+    }
+    byte[] data = null;
+    if (packetLength >= payloadDataStart) {
+      data = new byte[packetLength - payloadDataStart];
+      System.arraycopy(raw, ipOffset + payloadDataStart, data, 0, data.length);
+    }
+    return data;
+  }
+
+  private InetAddress getIPAddress(final boolean src) {
+    byte[] ipBuffer = getIpAddressBytes(src);
+    if (ipBuffer == null) {
+      return null;
+    }
+    try {
+      return InetAddress.getByAddress(ipBuffer);
+    } catch (UnknownHostException e) {
+      return null;
+    }
+  }
+
+  private byte[] getIpAddressBytes(final boolean src) {
+    int srcPos;
+    byte[] ipBuffer;
+    if (isIpV4Packet()) {
+      ipBuffer = new byte[4];
+      srcPos = src ? PacketConstants.IP4_SRC_OFFSET : PacketConstants.IP4_DST_OFFSET;
+    } else if (isIpV6Packet()) {
+      ipBuffer = new byte[16];
+      if (isRoutingV6) {
+        srcPos = src ? PacketConstants.IP6_SRC_OFFSET + 96 : PacketConstants.IP6_DST_OFFSET + 96;
+      } else {
+        srcPos = src ? PacketConstants.IP6_SRC_OFFSET : PacketConstants.IP6_DST_OFFSET;
+      }
+    } else if (isPPPoV6Packet()) {
+      ipBuffer = new byte[16];
+      srcPos = src ? PacketConstants.IP6_SRC_OFFSET + PacketConstants.PPPoV6_IP_OFFSET : PacketConstants.IP6_DST_OFFSET + PacketConstants.PPPoV6_IP_OFFSET;
+    } else {
+      return null;
```

Comment:
If there are unsupported types we will end up with a null value in the output. Is that what is desired? I tried with a pcap file from here: http://www.netresec.com/?page=MACCDC and verified that is so. That would be one of the reasons I feel we should at least document what might not be supported? Your call on this one. I am not an expert on the matter.